enhancements/network/upi-node-ip-selection.md

```diff
@@ -0,0 +1,213 @@
+---
+title: upi-node-ip-selection
+authors:
+  - "@pliu"
+reviewers: 
+  - "@racedo"
+  - "@makentenza"
+  - "@cybertron"
+  - "@mkowalski"
+approvers:
+  - "@danwinship"
+  - "@trozet"
+api-approvers: 
+  - "None"
+creation-date: 2023-02-08
+last-updated: 2023-02-08
+tracking-link:
+  - https://issues.redhat.com/browse/NP-687
+see-also:
+  - "/enhancements/network/ip-interface-selection"
+replaces:
+superseded-by:
+---
+
+# Node IP Selection UPI
+
+## Summary
+
+When Openshift is deployed on user-provisioned infrastructure, and there are
+more than one NIC available on the hosts. We need a reliable way to guarantee
+all the openshift components can use the correct IP/interface.
+
+## Motivation
+
+Today, we do not run the nodeip-configuration service on vSphere UPI. It means
+we don't have the mechanism to ensure IP/interface selection will return the
+same result for following components:
+
+- Node IP (Kubelet and CRIO)
+- configure-ovs
+- resolv-prepender
+- Keepalived
+- Etcd
+
+For other UPI platforms, although the nodeip-configuration service is enabled.
+It runs in a mode that it will only return the IP/interface with the default
+route. That is not always what a customer want. We need to have a generic way
+for all the UPI platforms for the node IP/interface selection.
```

Comment:
After reading this carefully, I have a general understanding of the technical problem, but the broader motivation is still not clear to me. What new functionality does this allow us or what problems does this allow us to avoid?

Reply:
I updated the motivation section. PTAL